src/init.cpp

```diff
@@ -1764,9 +1778,17 @@ bool AppInitMain(NodeContext& node)
 
     Discover();
 
-    // Map ports with UPnP
-    if (gArgs.GetBoolArg("-upnp", DEFAULT_UPNP)) {
-        StartMapPort();
+    // Map ports with NAT-PMP (preferable) or UPnP.
+    {
+        const bool natpmp = gArgs.GetBoolArg("-natpmp", DEFAULT_NATPMP);
+        const bool upnp = gArgs.GetBoolArg("-upnp", DEFAULT_UPNP);
+        if (natpmp && upnp) {
+            StartMapPort(MapPort::ANY_AVAILABLE);
```

Comment:
Suggest just making this `StartMapPort(bool use_upnp, bool use_natpmp)`

Reply:
Or better yet: Just check gArgs inside StartMapPort...

Reply:
Reworked in the latest push.